torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -267,6 +267,14 @@ c10::optional<Node*> tryMerge(
       getHeader(consumer),
       ":\n");
 
+  // Shortcut for Block code generation for single op
+  if (tensorexpr::getTEGenerateBlockCode() &&
```

Comment:
I am fine with this in the short term, but we should be committed to cleaning it up soon. At least add a TODO to yourself to remind you of that.

What I am objecting to is that this code is looking at a global flag to decide whether to generate Block code. This is generally pretty bad. We should only look at input_ device to pick whether this is the target device. SImilar to how CPU and GPU specific codegen is invoked. 

https://github.com/pytorch/pytorch/blob/6bdfd6ae1adb341f5bae2fc8bc987653b8a319ad/torch/csrc/jit/tensorexpr/kernel.cpp#L1392

Reply:
Thanks, this was really useful feedback. I added the TODO for now. I'll investigate adding the Block device type.